### PR TITLE
do not create StatefulSet pods when PVC is being deleted

### PR DESCRIPTION
**Pull request by andyxning**
Cherry pick of #93457 on release-1.19.

Pod with PVC will not be scheduled if the PVC is being deleted.
This can happen when the PVC has finalizers of storage plugins.

Such a pod becomes pending.  Unfortunately, after the finalizer
finishes and PVC is deleted, the pod remains pending forever.
The StatefulSet controller does nothing for this pending pod.

This commit prevents the StatefulSet controller from creating
such pods when PVC is to be deleted.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andyxning** [comment]
@kubernetes/release-managers 
/assign @tnozicka

**Review by hasheddan — COMMENTED** [review]
@andyxning would this also be appropriate to CP to 1.18?

**Comment by andyxning** [comment]
> @andyxning would this also be appropriate to CP to 1.18?

@hasheddan If it is ok, i will fire a PR to do this. But it seems that no one is answering this. :(

**Comment by andyxning** [comment]
@derekwaynecarr @mikedanese @janetkuo @cheftako @deads2k

**Comment by andyxning** [comment]
Ping @derekwaynecarr @mikedanese @janetkuo @cheftako @deads2k  @justaugustus @saschagrunert @hasheddan

**Comment by justaugustus** [comment]
> @andyxning: This cherry pick PR is for a release branch and has not yet been approved by [Release Managers](https://git.k8s.io/sig-release/release-managers.md).
> Adding the `do-not-merge/cherry-pick-not-approved` label.
> 
> To merge this cherry pick, it must first be approved (`/lgtm` + `/approve`) by the relevant OWNERS.
> 
> **AFTER** it has been approved by code owners, please ping the **kubernetes/release-managers** team in a comment to request a cherry pick review.
> 
> (For details on the patch release process and schedule, see the [Patch Releases](https://git.k8s.io/sig-release/releases/patch-releases.md) page.)

@andyxning -- This needs to be reviewed and approved by the [code owners](https://github.com/kubernetes/kubernetes/pull/98732#issuecomment-772755700) before it can be approved by Release Managers for cherry pick.

Please do not spam assign people.

/uncc @hasheddan 
/unassign @justaugustus @saschagrunert @hasheddan 
/assign @janetkuo @tnozicka

**Comment by ymmt2005** [comment]
Looks good to me.  I don't have the rights to give /lgtm or /approve, though.

**Comment by andyxning** [comment]
> Looks good to me. I don't have the rights to give /lgtm or /approve, though.

Thanks @ymmt2005 

/cc @liggitt @hasheddan Correct me if i wrongly ping someone. :(

**Comment by liggitt** [comment]
@janetkuo @tnozicka should review/approve as the `master` PR reviewers and sig representatives

**Comment by andyxning** [comment]
IMHO, this is an important bugfix and needs to be backport. PTAL. @janetkuo @tnozicka

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm

This is an important fix for StatefulSet PVC handling.

**Pull request merged**
